internal/graph/storagewrapper.go

```diff
@@ -449,13 +449,6 @@ func (c *cachedIterator) Stop() {
 		defer c.wg.Done()
 		defer c.iter.Stop()
 
-		// if cache is already set, we don't need to drain the iterator
```

Comment:
The cache may be present but invalid.  Thus, we still need to drain the iterator.

Reply:
we kept the above logic but added in condition on invalidation